tsl/src/continuous_aggs/common.c

```diff
@@ -415,11 +418,16 @@ caggtimebucket_validate(CAggTimebucketInfo *tbinfo, List *groupClause, List *tar
 						 errhint("Use an immutable expression as first argument to the time bucket "
 								 "function.")));
 			}
+
+			tbinfo->bf->bucket_function = fe->funcid;
+			tbinfo->bf->bucket_time_based =
```

Comment:
Is this definitely time based func at this point?

Reply:
It can be depending on the width argument type!